sdk/identity/identity/review/identity.api.md

```diff
@@ -155,9 +155,13 @@ export class InteractiveBrowserCredential implements TokenCredential {
     }
 
 // @public
-export interface InteractiveBrowserCredentialOptions extends TokenCredentialOptions {
+export interface InteractiveBrowserCredentialBrowserOptions extends InteractiveBrowserCredentialCommonOptions {
+    clientId: string;
+}
+
+// @public
+export interface InteractiveBrowserCredentialCommonOptions extends TokenCredentialOptions {
```

Comment:
I'm a bit torn about this update, I didn't want to completely change the NodeJS options bag, but I wanted to make the client ID required for the browser.

Reply:
why not simplify this and make `clientId` mandatory when run in node as well? From what I recall, we are indeed regretting adding a default clientId that maps to Azure CLI. Since we are making breaking changes, why not use this opportunity to do the right thing?

Reply:
`InteractiveBrowserCredential` for non-browser environments also uses the CLI client ID in other languages! I think we should keep that working and aligned.

Reply:
It may be worth discussing aligning the credential across node and browser in the same language rather than across languages.

Reply:
I disagree. InteractiveBrowserCredential needs to function quite differently when executed in the browser, so I don't think you can get around having different behavior between node and web.

Reply:
Wouldnt that difference be in the underlying implementation and not as much in how we instantiate the class?

I am going under the assumption that if given the chance we would have not gone with using the Azure CLI clientId for InteractiveBrowserCredential in any language. Since the current changes we are making will need us to bump the major version, why not take the opportunity to do the right thing for InteractiveBrowserCredential in node and make clientId mandatory? If the above mentioned assumption is wrong, then I take back my case

Reply:
> Wouldnt that difference be in the underlying implementation and not as much in how we instantiate the class?

The difference is unfortunately deeper than just an implementation detail. When executing in the browser the redirect is handled differently then when we pop the browser from a node and redirect to local host. To accomplish this a different AAD application type is needed. I would say that the "right" thing to do if we want to not break would be to create a new first party app which can serve as a default when executing in the browser. However there is a process to create this and it can take some time.